system/include/emscripten/em_macros.h

```diff
@@ -14,30 +14,3 @@
 #else
 #define EM_IMPORT(NAME)
 #endif
-
-/*
- * EM_JS_DEPS: Use this macro to declare indirect dependencies on JS symbols.
- * The first argument is just unique name for the set of dependencies.  The
- * second argument is a C string that lists JS library symbols in the same way
- * they would be specified in the DEFAULT_LIBRARY_FUNCS_TO_INCLUDE command line
- * setting.
- *
- * For example, if your code contains an EM_ASM or EM_JS block that make use of
- * the allocate and stackSave JS library functions then you might write this in
- * your library source code:
- *
- *   EM_JS_DEPS(mylib_dep, "$allocate,$stackSave");
- *
- * The emscripten linker will then pick this up and make sure those symbols get
- * included in the JS support library.
- *
- * Dependencies declared in this way will be included if-and-only-if the object
- * file (translation unit) in which they exist is included by the linker, so
- * it makes sense co-locate them with the EM_JS or EM_ASM code they correspond
- * to.
- */
-#define EM_JS_DEPS(tag, deps)            \
-  EMSCRIPTEN_KEEPALIVE                    \
-  __attribute__((section("em_lib_deps"))) \
-  __attribute__((aligned(1)))             \
-  char __em_lib_deps_##tag[] = deps;
```

Comment:
I think the reason I put them in em_macros.h was so they could be used then only including `em_asm.h`.. since this is useful for both EM_JS and EM_ASM users.

Not a big deal I guess.

Reply:
Ah. I figured this way it was easier to reuse the `_EM_JS_CPP_BEGIN` helper instead of copy-pasting, plus the name of the macro seems to imply it belongs alongside `EM_JS`.

I don't mind moving back + copying the helper if that's preferable.

Reply:
You can move the `_EM_JS_CPP_BEGIN` macros in to `em_macros` and just call them `_EM_CPP_BEGIN` maybe?

Reply:
Hmm.. `_EM_JS_CPP_BEGIN` is strangely names since it marks the start of C declaration, no C++.

How about `_EM_BEGIN_CDECLS` and ``_EM_END_CDECLS`

Reply:
Hm we could also simplify them to just `_EM_CDECL` since we don't really need block - we only apply it on per-item basis.

I'll try that.